### PR TITLE
Adding tox environment for testing asset validation

### PR DESCRIPTION
**Pull request by sebworks**
Adding tox environment for testing asset validation

## Changes

- Modified `frontend.sh` to add a `dry-run` function.

## Testing

1. Run `tox -e validate-assets -v` and hope you don't see `[object object]`.
2. You can run `tail -f ` on the log file to check the dependency build cycle.

## Notes

-

## Todos

-

## Checklist

* [ ] PR has an informative and human-readable title
* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
* [ ] Passes all existing automated tests
* [ ] Any *change* in functionality is tested
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [ ] Reviewers requested with the [Reviewer tool](https://help.github.com/articles/about-pull-request-reviews/) :arrow_right:

### CODE REVIEW

File: frontend.sh [hunk 1]
```diff
@@ -93,6 +93,17 @@ build() {
   fi
 }
 
+# Run tasks to build all project assets for production distribution.
+dry_run() {
```
Comment (willbarton) [inline frontend.sh]: I'm not sure `dry_run` is the best descriptor for what this does. `validate_static_files` maybe?

File: frontend.sh [hunk 2]
```diff
@@ -93,6 +93,17 @@ build() {
...
+  set DJANGO_SETTINGS_MODULE=cfgov.settings.production
+  set DJANGO_STATIC_ROOT=$(pwd)/collectstatic
+  pip install -r requirements/optional-public.txt
+  pip install -r requirements/optional-private.txt
```
Comment (willbarton) [inline frontend.sh]: It occurs to me that we *might* want to do this in a tox environment to avoid muddying up someone's working env.

Comment (sebworks) [reply frontend.sh]: yeah, how would you do that outside a testing environment?

Comment (willbarton) [reply frontend.sh]: Create a testing environment in tox.ini :) 

Unfortunately to do so would require `tox` to be installed, which I don't think we can expect to have when we get to `frontend.sh`. 

Happy to help figure this out if you like.

Comment (sebworks) [reply frontend.sh]: ok, I've added a script to tox.ini but I'm now getting a `No module named MySQLdb` error. I'm not sure why MySQLdb would be necessary when running collectstatic. Also, it's hecka slow and the `-vv` flag doesn't work as expected unless we upgrade to tox 2.9.

Comment (sebworks) [reply frontend.sh]: ok, I just added mysql as a dependency. Should be good to go now.

File: tox.ini [hunk 3]
```diff
@@ -87,4 +87,23 @@ setenv=
     {[testenv]setenv}
     DJANGO_SETTINGS_MODULE=cfgov.settings.test
 commands=
-    ./manage.py runscript missing_migrations 
+    ./manage.py runscript missing_migrations
+
+[testenv:validate-assets]
+# Invoke with: tox -e validate-assets
+# Ensure all assets are generated without error.
+envdir={[testenv]envdir}-validate-assets
```
Comment (willbarton) [inline tox.ini]: I would leave this as `envdir={[testenv]envdir}` like the other envs, then tox doesn't have to set up a brand new one just for this. That setup time is part of why it takes so long.

File: frontend.sh (hunk 1 above)
Comment (willbarton) [inline frontend.sh]: The tox environment works beautifully. Can we remove this function, or have it call `tox -e validate-assets` (and rename it if we're keeping it)?

File: tox.ini [hunk 4]
```diff
@@ -87,4 +87,23 @@ setenv=
...
+changedir={toxinidir}
+recreate=False
+deps=
+    -r{toxinidir}/requirements/optional-public.txt
+    -r{toxinidir}/requirements/optional-private.txt
+    {[testenv]deps}
+    mysql
```
Comment (willbarton) [inline tox.ini]: If you add `-r{toxinidir}/requirements/mysql.txt`, you'll be including the `mysql` package we install for local and production.

File: tox.ini (hunk 3 above)
Comment (sebworks) [reply tox.ini]: fixed in https://github.com/cfpb/cfgov-refresh/pull/3537/commits/70d709ce9ce0ce058da1cfa9761914817d8341ca

File: tox.ini (hunk 4 above)
Comment (sebworks) [reply tox.ini]: fixed in https://github.com/cfpb/cfgov-refresh/pull/3537/commits/70d709ce9ce0ce058da1cfa9761914817d8341ca

File: frontend.sh (hunk 1 above)
Comment (sebworks) [reply frontend.sh]: fixed in https://github.com/cfpb/cfgov-refresh/pull/3537/commits/70d709ce9ce0ce058da1cfa9761914817d8341ca.

**Pull request merged**